pyscf/df/df_jk.py

```diff
@@ -132,28 +132,32 @@ def reset(self, mol=None):
 
     def get_jk(self, mol=None, dm=None, hermi=1, with_j=True, with_k=True,
                omega=None):
+        assert (with_j or with_k)
         if dm is None: dm = self.make_rdm1()
         if not self.with_df:
             return super().get_jk(mol, dm, hermi, with_j, with_k, omega)
 
         with_dfk = with_k and not self.only_dfj
-        if isinstance(self, scf.ghf.GHF):
-            def jkbuild(mol, dm, hermi, with_j, with_k, omega=None):
-                vj, vk = self.with_df.get_jk(dm.real, hermi, with_j, with_k,
-                                             self.direct_scf_tol, omega)
-                if dm.dtype == numpy.complex128:
-                    vjI, vkI = self.with_df.get_jk(dm.imag, hermi, with_j, with_k,
-                                                   self.direct_scf_tol, omega)
-                    if with_j:
-                        vj = vj + vjI * 1j
-                    if with_k:
-                        vk = vk + vkI * 1j
-                return vj, vk
-            vj, vk = scf.ghf.get_jk(mol, dm, hermi, with_j, with_dfk,
-                                    jkbuild, omega)
+        if with_j or with_dfk:
```

Comment:
I think there may problem here with the `else` clause: the call is to
```
                vj, vk = self.with_df.get_jk(dm, hermi, with_j, with_dfk,
                                            self.direct_scf_tol, omega)
```
and I wonder if this plays nicely for the various cases. I don't know the logic of the code well enough to review this TBH

Reply:
There are no changes to this `if/else` structure except that this part of the code is only invoked if Coulomb integrals are calculated or density-fitted exchange integrals are available. The call to `self.with_df.get_jk()` will only cause issues AFAIK when `with_j` and `with_dfk` are both `False` which should not happen due to the added `if` clause and since either `with_j` and `with_k` are asserted to be `True`.